interview-questions.md

```diff
@@ -8,89 +8,104 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. Why would you use the method `super()`?
 
   Your answer:
-
+    super() is used as a method within a class. This allows the class to access the variables and data from the other class that was super()'d in.
   Researched answer:
+    super() is a method that is used in class inheritance. After a child class is extended from its parent class, super() is evoked within its constructor() to  inherit the attributes from its parent. You can also use super() to call on methods from the parent class.
 
 
 
 2. What is the DOM? What is the virtual DOM?
 
   Your answer:
-
+    The DOM is something we can manpipulate as coders to change the layout of the webpage.
   Researched answer:
+    DOM stands for Document Object Mode. It's a programming interface that treats HTML or XML documents like a tree structure where each node acts as an object in the document. Manipulating the DOM directly can be cumbersome because the page must render after every change. The the virtual DOM is React's local copy of the HTML DOM. Having a virtual DOM allows React to do its computations while skipping the real DOM's operations.
 
 
 
 3. What is a higher-order function?
 
   Your answer:
-
+    A higher-order function is a function that uses a callback function within its own lines of code. Callback functions include .map(), .filter(), and .forEach(). These built-in methods are useful to reduce code. They perform some sort of iteration, so you don't have to create for-loops and save some memory in run-time and production. 
   Researched answer:
-
+    A higher-order function is a function that uses a callback function within its function body or returns a function, as opposed to first order functions that do neither. Callback functions include .map(), .filter(), and .forEach(). These built-in methods are useful to reduce code. They perform some sort of iteration, so you don't have to create for-loops and save some memory and efficiency in run-time and production. 
 
 
 4. What is JSX?
 
   Your answer:
+    JSX is a the bridge between HTML and Javascript. JSX looks like HTML, but it works within JavaScript folders. It's used in React projects, and you can find them within the return in the render in the component definition. One difference that it has from HTML is that it uses "className" instead of "class".
```

Comment:
Pretty good answer! Something to note is that JSX is NOT specific to just React. It's utilized by React. Additionally, the 'className' note is really great! It might be worth noting that it is also not just a React thing, but its used as className due to 'class' being a protected word in Javascript, which is what JSX gets compiled down to.